src/app/types.ts

```diff
@@ -1,5 +1,27 @@
+import type { WordList } from "@common/types";
 export interface DictionarySource {
   name: string;
   size: number;
   type: string;
 }
+export interface StoredWordList {
+  /** Autogenerated primary key */
+  id?: number;
+
+  /**
+   * Typically the filename of the uploaded dictionary source.
+   */
+  name: string;
+  /**
+   * The actual contents of said file.
+   */
+  wordlist: WordList;
+  /**
+   * Size of the file in bytes.
+   */
+  size?: number;
+  /**
+   * File extension.
+   */
+  type?: string;
+}
```

Comment:
This looks like a copy-paste from`@worker/storage-types`? It's better to maintain a single copy of this class, and that copy should be in `@common/types` — so please move the interface to `@common/types`!

Is the `id?` required information for the UI? Or is it exclusive to the storage layer (Dexie/IndexedDB)? In which case, it should not be in the `@common/types` or even `@app/types` file!

You can do something like this. In `@common/types`:

```typescript
export interface WordlistSource {
  /* ... all properties EXCEPT id */
}
```

Then in `@worker/storage-types`:

```typescript
export interface StoredWordlist extends WordlistSource {
  /** Autogenerated primary key */
  id?: number;
}
```

Reply:
I think this would work the best! I'll update the PR accordingly. Thanks!